swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: ballerina-create-your-first-connector-left-nav-pages-swanlake
+title: Create your first connector with Ballerina
+description: step-by-step guide to create Ballerina connectors using OpenAPI specifications.
+keywords: ballerina, connector, library, REST, API, openAPI, Ballerina Central
+permalink: /learn/create-your-first-connector-with-ballerina/
+active: create-your-first-connector-with-ballerina
+intro: This guide walks you through creating your first Ballerina connector using an OpenAPI specification.
+---
+
+## Introduction
+
+Ballerina is a programming language designed for smooth integration, offering a large library of pre-built connectors. These connectors are special packages that make it easy to communicate with external services, usually through REST APIs. By using connectors, developers can quickly integrate third-party services into their Ballerina applications without having to worry about the technical details of API interactions.
+
+Along with its powerful library ecosystem, Ballerina also allows developers to easily create, share, and manage their own custom connectors. These connectors can be published on Ballerina Central, making them available for the entire community to use and reuse in their projects.
+
+In this guide, we'll walk you through how to generate your first Ballerina connector using an OpenAPI specification. This is one of the fastest and easiest ways to build connectors, helping you quickly integrate external services into your Ballerina projects.
+
+## Set up the prerequisites
+
+Before we begin, make sure you have:
+
+1. A basic understanding of Ballerina SwanLake and the [latest version installed](https://ballerina.io/downloads/).
+2. An OpenAPI specification of the API for which you’re building the connector, along with API credentials (e.g., Twitter Developer account).
+3. A GitHub account, and Git installed locally.
```

Comment:
Is this and getting started with a GH repo really required? If at all, I feel like it should come at the end, since it is not mandatory.

Reply:
Although this is not required for publishing, having the recommended project structure is required to house all the standard components of a Ballerina connector (tests, examples, CI/CD workflows, build scripts, other docs), keeping the standard connector structure will also make it easier for the users to contribute the same project as a Ballerina official connector in the future. So we thought of going with the repo approach as a required step. wdyt?

Reply:
Yeah the ideal flow would be to have a standard github project template for connectors and, ask from the users to just clone it. 

But since Github project template doesn't support template variables, we can't use that path. so we have to have to ask from the users to create their own empty repo, clone it, and copy the content from our provided template.

Reply:
I would still say this is optional and only required if you want to push the changes to GitHub.